magenta/music/musicxml_parser.py

```diff
@@ -0,0 +1,678 @@
+
+  def gettempos(self):
+    """Return a list of all tempos in this score.
+    If no tempos are found, create a default tempo of 120 qpm.
+    """
+    tempos = []
+    #for part in self.parts:
+    part = self.parts[0] # Use only first part
+    for measure in part.measures:
+      for tempo in measure.tempos:
+        tempos.append(tempo)
+
+    # If no tempos, add a default of 120 at beginning
+    if len(tempos) == 0:
+      tempo = Tempo(self.state)
+      tempo.qpm = self.state.tempo
+      tempo.time_position = 0
+      tempos.append(tempo)
+
+    return tempos
+
+
+class ScorePart(object):
+  """"Internal representation of a MusicXML <score-part>.
+  A <score-part> element contains MIDI program and channel info
+  for the <part> elements in the MusicXML document.
+
+  If no MIDI info is found for the part, use the next available
+  MIDI channel and default to the Grand Piano program (MIDI Program #1)
+  """
+  def __init__(self, state, xml_score_part=None):
+    self.xml_score_part = xml_score_part
+    self.part_name = ""
+    self.midi_channel = 1
+    self.midi_program = 1
+    self.state = state
+    if xml_score_part != None:
+      self.parse()
+
+  def parse(self):
+    """Parse the <score-part> element to an in-memory representation"""
+    if self.xml_score_part.find("part-name") != None:
+      self.part_name = self.xml_score_part.find("part-name").text
+
+    xml_midi_instrument = self.xml_score_part.find("midi-instrument")
+    if xml_midi_instrument != None and \
+      xml_midi_instrument.find("midi-channel") != None and \
+      xml_midi_instrument.find("midi-program") != None:
+      self.midi_channel = int(xml_midi_instrument.find("midi-channel").text)
+      self.midi_program = int(xml_midi_instrument.find("midi-program").text)
+    else:
+      # If no MIDI info, increment MIDI channel and use default program
+      self.midi_channel = self.state.midi_channel + 1
+      self.state.midi_channel = self.midi_channel
+      self.midi_program = 1
+
+  def __str__(self):
+    score_str = "ScorePart: " + self.part_name
+    score_str += ", Channel: " + str(self.midi_channel)
+    score_str += ", Program: " + str(self.midi_program)
+    return score_str
+
+
+class Part(object):
+  """Internal represention of a MusicXML <part> element"""
+  def __init__(self, xml_part, score_part, state):
+    self.xml_part = xml_part
+    self.score_part = score_part
+    self.measures = []
+    self.transposes = False
+    self.state = state
+    self.parse()
+
+  def parse(self):
+    """Parse the <part> element"""
+
+    # Reset the time position when parsing each part
+    #CURRENT_TIME_POSITION = 0
+    self.state.time_position = 0
+    self.state.midi_channel = self.score_part.midi_channel
+    self.state.midi_program = self.score_part.midi_program
+    self.state.transpose = 0
+
+    xml_measures = self.xml_part.findall("measure")
+    for child in xml_measures:
+      measure = Measure(child, self, self.state)
```

Comment:
For some reason, I thought it would be useful to indicate that the Part transposes, and Measure was mutating that flag on the Part since transposition is identified at the measure level in MusicXML (see my comments above for an example use case).

I am not using this attribute on the Part object at all, so it looks like it was just old code that needs to be cleaned up. I have removed Part.transposes and am now no longer passing the Part into the Measure constructor. Fixed in commit c581826.